Packs/RSANetWitness_v11_5/Integrations/RSANetWitness115/RSANetWitness115.py

```diff
@@ -0,0 +1,1120 @@
+    agent_id = args.get('agent_id')
+    service_id = args.get('service_id', SERVICE_ID)
+    allow_dns_only = args.get('allow_dns_only_by_system')
+    exclusion_list = create_exclusions_list(args.get('exclusion_list')) if args.get('exclusion_list') else None
+    comment = args.get('comment')
+
+    client.endpoint_update_exclusions_request(agent_id, service_id, allow_dns_only, exclusion_list, comment)
+    command_results = CommandResults(
+        readable_output=f'Isolate update exclusions request, for Host {agent_id} ,sent successfully'
+    )
+
+    return command_results
+
+
+def endpoint_isolation_remove_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    agent_id = args.get('agent_id')
+    service_id = args.get('service_id', SERVICE_ID)
+    comment = args.get('comment')
+    allow_dns_only = args.get('allow_dns_only_by_system')
+
+    client.endpoint_isolation_remove_request(agent_id, service_id, allow_dns_only, comment)
+    command_results = CommandResults(
+        readable_output=f'Isolate remove request, for Host {agent_id} ,sent successfully'
+    )
+
+    return command_results
+
+
+def fetch_incidents(client: Client, fetch_time, fetch_limit):
+    fetch_limit = int(fetch_limit)
+    last_run = demisto.getLastRun()
+    if last_run and last_run.get('timestamp'):
+        timestamp = last_run.get('timestamp')
+        last_fetched_ids = last_run.get('last_fetched_ids', [])
+    else:
+        last_fetch = dateparser.parse(fetch_time)
+        # convert to ISO 8601 format and add Z suffix
+        timestamp = last_fetch.isoformat() + 'Z'
+        last_fetched_ids = []
+
+    total_items = get_incidents(client, timestamp, fetch_limit, last_fetched_ids)
+    incidents: List[Dict] = []
+    new_ids = []
+    for item in total_items:
+        new_ids.append(item.get('id'))
+        incident = {"name": f"RSA NetWitness 11.5 {item.get('id')}",
+                    "occurred": item.get('created'),
+                    "rawJSON": json.dumps(item)}
+        # items arrived from last to first - change order
+        incidents.insert(0, incident)
+
+    new_last_run = incidents[-1].get('occurred') if incidents else timestamp
+    if is_new_run_time_equal_last_run_time(timestamp, new_last_run):
+        new_ids.extend(last_fetched_ids)
+
+    demisto.setLastRun({"timestamp": new_last_run, "last_fetched_ids": new_ids})
+    return incidents
+
+
+def get_incidents(client, timestamp, fetch_limit, last_fetched_ids):
+    page_size = 100
+    response = client.list_incidents_request(None, timestamp, page_size, 0)
+    if not response.get('items'):
+        return []
+
+    page_number = response.get('totalPages') - 1
+    total = 0
+    total_items: List[Dict] = []
+    while total < fetch_limit and page_number >= 0:
+        response = client.list_incidents_request(None, timestamp, page_size, page_number)
+        items = response.get('items', [])
+        new_items = remove_duplicates_for_fetch(items, last_fetched_ids)
```

Comment:
Why do we need a duplicate mechanism if we can set the 'until' to be the time we started the loop?

Reply:
You're right, removed this check and added the until argument